src/operator/mshadow_op.h

```diff
@@ -676,15 +676,19 @@ struct mod {
       return DType(0);
     } else if (b < DType(0)) {
       if (a < DType(0)) {
-        return DType(-::fmod(-a, -b));
+        return DType(-::fmod(-static_cast<double>(a), -static_cast<double>(b)));
```

Comment:
In retrospect, I could have implemented it with `fmodf` too, though the cost is more specializations, which complicates the codes.